### PR TITLE
Natasha/class fix

### PR DESCRIPTION
**Pull request by nkoussa**
Fixes errors when using classification metrics, double checked with a regression model

### CODE REVIEW

File: improvelib/metrics.py [hunk 1]
```diff
@@ -50,7 +54,8 @@ def str2Class(str) -> Any:
 
 def compute_metrics(y_true: np.ndarray,
                     y_pred: np.ndarray,
-                    metric_type: str
+                    metric_type: str, 
+                    y_prob = None
```
Comment (jonesse3) [inline improvelib/metrics.py:58]: Don't forget to add y_prob to the Args section of the docstrings

Comment (nkoussa) [reply improvelib/metrics.py:58]: done, thanks

File: improvelib/metrics.py [hunk 2]
```diff
@@ -227,27 +249,31 @@ def recall(y_true: np.ndarray, y_pred: np.ndarray) -> float:
...
 
 
-def auc(y_true: np.ndarray, y_pred: np.ndarray) -> float:
+def roc_auc(y_true: np.ndarray, y_prob: np.ndarray) -> float:
     """Compute Receiver Operating Characteristic AUC.
 
     Args:
         y_true (np.ndarray): True values to predict.
-        y_pred (np.ndarray): Predictions made by the model.
+        y_pred (np.ndarray): Target scores made by the model.
```
Comment (jonesse3) [inline improvelib/metrics.py]: Need to change to y_prob if changing it in the function

File: improvelib/metrics.py [hunk 3]
```diff
@@ -227,27 +249,31 @@ def recall(y_true: np.ndarray, y_pred: np.ndarray) -> float:
...
 
 
-def aupr(y_true: np.ndarray, y_pred: np.ndarray) -> float:
+def aupr(y_true: np.ndarray, y_prob: np.ndarray) -> float:
     """Compute Precision-Recall curve AUC.
 
     Args:
         y_true (np.ndarray): True values to predict.
-        y_pred (np.ndarray): Predictions made by the model.
+        y_pred (np.ndarray): Target scores made by the model.
```
Comment (jonesse3) [inline improvelib/metrics.py]: Same here: Need to change to y_prob if changing it in the function

File: improvelib/metrics.py [hunk 4]
```diff
@@ -68,7 +73,7 @@ def compute_metrics(y_true: np.ndarray,
     scores = {}
 
     if metric_type == "classification":
-        metrics = ["acc", "recall", "precision", "f1", "auc", "aupr"]
+            metrics = ["mse", "acc", "recall", "precision", "f1", "kappa", "bacc"]
```
Comment (jonesse3) [inline improvelib/metrics.py]: Is one of the classification models using mse?

File: improvelib/metrics.py (hunk 3 above)
Comment (nkoussa) [reply improvelib/metrics.py]: oops yeah

File: improvelib/metrics.py (hunk 4 above)
Comment (nkoussa) [reply improvelib/metrics.py]: no put it's in compute_performance_scores in utils.py so it errors if it's not there

Comment (jonesse3) [reply improvelib/metrics.py]: Let's use log loss then:

```
from sklearn.metrics import log_loss

def logloss(y_true: np.ndarray, y_prob: np.ndarray) -> float:
    """Compute Log Loss (Cross-Entropy Loss) for classification.

    Args:
        y_true (np.ndarray): True binary or multi-class labels.
        y_prob (np.ndarray): Target scores made by the model (probabilities).

    Returns:
        float: The computed Log Loss.
    """
    return log_loss(y_true, y_prob)
```

Comment (jonesse3) [reply improvelib/metrics.py]: We can skip for now if none of the classification models are printing out this result.

**Review by jonesse3 — APPROVED** [review]
